app/code/Magento/CatalogSearch/Model/ResourceModel/Fulltext/Collection.php

```diff
@@ -265,7 +265,7 @@ public function addFieldToFilter($field, $condition = null)
 
         $this->getSearchCriteriaBuilder();
         $this->getFilterBuilder();
-        if (!is_array($condition) || !in_array(key($condition), ['from', 'to'])) {
+        if (!is_array($condition) || !in_array((string)key($condition), ['from', 'to'])) {
```

Comment:
I can confirm that the suggested change by @v-r along with this pull request do solve the problem. We need this fix which we've made on our local install. @allanpaiste any chance you can update this one line and submit the PR again? I could make a new request but I'd just be copying your work!